### PR TITLE
Migrate Iceberg Catalog tests to JUnit

### PR DESCRIPTION
**Pull request by findepi**

### CODE REVIEW

**Review by losipiuk — APPROVED** [review]

File: plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/catalog/BaseTrinoCatalogTest.java [hunk 1]
```diff
@@ -36,7 +36,7 @@
 import org.apache.iceberg.Table;
 import org.apache.iceberg.expressions.Expressions;
 import org.apache.iceberg.types.Types;
-import org.testng.annotations.Test;
+import org.junit.jupiter.api.Test;
```
Comment (findinpath) [inline plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/catalog/BaseTrinoCatalogTest.java:40]: Replaces as well `assertEquals` and `assertNotEquals` with

```
import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertNotEquals;
```

File: plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/catalog/rest/TestTrinoRestCatalog.java [hunk 2]
```diff
@@ -61,13 +62,15 @@ protected TrinoCatalog createTrinoCatalog(boolean useUniqueTableLocations)
         return new TrinoRestCatalog(restSessionCatalog, new CatalogName(catalogName), NONE, "test", useUniqueTableLocations);
     }
 
+    @Test
```
Comment (findinpath) [inline plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/catalog/rest/TestTrinoRestCatalog.java:65]: Unrelated to this PR - do we have any plans to figure out "forgotten" test overrides which are missing now `@Test` ?

**Review by findinpath — APPROVED** [review]

File: plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/catalog/glue/TestTrinoGlueCatalog.java [hunk 3]
```diff
@@ -35,7 +35,7 @@
 import io.trino.spi.security.PrincipalType;
 import io.trino.spi.security.TrinoPrincipal;
 import io.trino.spi.type.TestingTypeManager;
-import org.testng.annotations.Test;
+import org.junit.jupiter.api.Test;
```
Comment (findinpath) [inline plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/catalog/glue/TestTrinoGlueCatalog.java:38]: Replace testng `assertEquals` with 

```
import static org.junit.jupiter.api.Assertions.assertEquals;
```

File: plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/catalog/BaseTrinoCatalogTest.java (hunk 1 above)
Comment (findepi) [reply plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/catalog/BaseTrinoCatalogTest.java:40]: it's better to separate changes in test framework structure from test contents

File: plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/catalog/rest/TestTrinoRestCatalog.java (hunk 2 above)
Comment (findepi) [reply plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/catalog/rest/TestTrinoRestCatalog.java:65]: there is automated check for that now.

File: plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/catalog/glue/TestTrinoGlueCatalog.java (hunk 3 above)
Comment (findepi) [reply plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/catalog/glue/TestTrinoGlueCatalog.java:38]: https://github.com/trinodb/trino/pull/19525#discussion_r1371552309

**Pull request merged**